### PR TITLE
Some Coverity Fixes

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Some Coverity fixes that hopefully don't break things and don't create new warnings. Plus one leak fix(?). Untested.

### CODE REVIEW

File: core/picture.c [hunk 1]
```diff
@@ -123,13 +123,13 @@ static bool dive_check_picture_time(const struct dive *d, timestamp_t timestamp)
  * The caller is responsible for actually adding the picture to the dive.
  * If no appropriate dive was found, no picture is created and NULL is returned.
  */
-struct picture *create_picture(const char *filename, int shift_time, bool match_all, struct dive **dive)
+struct picture *create_picture(const char *filename, offset_t shift_time, bool match_all, struct dive **dive)
```
Comment (dirkhh) [inline core/picture.c]: I really wonder if this is a valid assumption.

One of the "typical" cases with incorrect dates appears to be an incorrect "Unix epoch" date - so January 1970.
So yeah, that's 52 years.

Now, I don't know if the code would be broken in different ways for that case - but I just a few weeks ago stumbled across pictures that claimed to be from Dec 31 1969 (thanks, time zones)...

**Review by dirkhh — COMMENTED** [review]
I don't know if the code would work - I guess I should try it.
But at least I disagree with the commit message, if nothing else.

**Comment by bstoeger** [comment]
Ok, dropped first commit. I have no time to look closer into that.

**Comment by dirkhh** [comment]
> Ok, dropped first commit. I have no time to look closer into that.

works for me.

"I have no time" is my middle name right now...

(the tooling for the Ubuntu builds stumbles if I do two merges to quickly together... so I'll wait a few minutes and merge then)

**Pull request merged**